src/protocol/commands/_vFile_pread.rs

```diff
@@ -0,0 +1,28 @@
+use super::prelude::*;
+
+#[derive(Debug)]
+pub struct vFilePread {
+    pub fd: usize,
```

Comment:
This corresponds to a target's `c_int` type, which is usually an `i32`. If we wanted to be super correct, we'd plumb through some kind of Arch-level option to specify the type of `c_int`, but that's pretty high effort, so using i32 should suffice.

(this applies to all other instances of `fd` in the other packets)

Note that this also complicated parsing a bit, since it means you'd have to handle negative hex-encoded numbers as well... You might want to create a new `decode_hex_int` helper to encapsulate this logic.

Reply:
oh, actually, it seems the spec defines the size of `int` when used as part of remote syscalls:

https://sourceware.org/gdb/onlinedocs/gdb/Integral-Datatypes.html#Integral-Datatypes

I think you'll find the parent page to be useful as well. notably, it specifies the layout of the "stat" structure, which will be needed to implement `fstat`:

https://sourceware.org/gdb/onlinedocs/gdb/Protocol_002dspecific-Representation-of-Datatypes.html#Protocol_002dspecific-Representation-of-Datatypes

Reply:
According to https://sourceware.org/gdb/onlinedocs/gdb/open.html#open, `open returns the new file descriptor or -1 if an error occurred.`
`fd` won't be negative. So should this type be `Option<NonZeroUsize>`?

Reply:
My hesitation is that `Option<u32>` would be 16 bytes, whereas `i32` is just 8 bytes...

I've thought about this for a while, and while your point that the fd _shouldn't_ ever be negative, I feel like I'd rather take the conservative route of parsing the fd into a i32 directly, and punting any error handling down to the handler implementation.

Reply:
If we implement `HostIoErrno` mentioned below, it means that the return value won't be negative except `-1`.

Reply:
Ahh, this is true...

Alright, in that case, I'd use `HostIoError<u32, Self>` as the return value.